### DISCUSSION TITLE
Color Management

### DISCUSSION THREAD

**Discussion by anuejn** [slint-ui/slint#4988] [Ideas]
Hey,
First of all, I would like to thank you for making this nice UI framework :).

I was considering to use slint to build a raw video player / processing tool. For this, I would like to be able to display images in a wide gamut colour space (i.e. DCI-P3) and HDR images. I am especially interested in doing that on macOS (since that is the only platform where I have hardware that can actually do this). 

Am I right assuming, that all the slint code is currently assuming everything to be in sRGB?
If so, It would be very cool to be able to create Images in other colour spaces and with floating point formats :).

**Comment by tronical** [comment]
It's a little worse than that, unfortunately :)

The software renderer blending routines and the color interpolation code (for color animation, or darker, ligher, mix function) assumes linear color space (not sRGB)  which is actually a bug, since the input (images and html colors) and output are indeed in sRGB.

The same goes for the Skia and FemtoVG renderers. I think the former we could reconfigure for sRGB, as well as the OpenGL context for the latter.

Yes, support for additional color spaces would be very desirable.

**Reply by owiofwm2i** [reply]
Hi, any update? Does Slint support HDR images to show up now? I saw the image still in 8-bit RGB, but Skia seems to support BGRA10_XR, don't know if it can be used.

**Reply by tronical** [reply]
Perhaps an intermittent solution would be to expose the (changing) skia_safe APIs with an image conversion function. That way you can use Skia‘s types directly, but you also need to adapt to changes in the API.
